client/me/sidebar/index.jsx

```diff
@@ -47,7 +48,11 @@ const MeSidebar = React.createClass( {
 		if ( isEnLocale && ! config.isEnabled( 'desktop' ) ) {
 			redirect = '/?apppromo';
 		}
-		userUtilities.logout( redirect );
+
+		this.props.logoutUser( redirect ).then( ( { redirect_to } ) => {
+			user.clear( () => location.href = redirect_to || '/' );
+		} );
```

Comment:
Reloading the current page is probably the only option for a failed logout - either the server couldn't load the the current user, or the nonce was invalid.